crucible-llvm/src/Lang/Crucible/LLVM/MemModel/Generic.hs

```diff
@@ -645,7 +647,12 @@ readMemInvalidate ::
   SymBV sym w   {- ^ The length of the set region        -} ->
   (StorageType -> LLVMPtr sym w -> ReadMem sym (PartLLVMVal sym)) ->
   ReadMem sym (PartLLVMVal sym)
-readMemInvalidate sym w end mop (LLVMPointer blk off) tp d msg sz readPrev =
+readMemInvalidate sym w end mop (LLVMPointer blk off) tp d msg sz readPrev
```

Comment:
Sadly, this is too easy to be correct, I think.  I think this has to be pushed down into the `InRange` cases of the `subFn` functions below, which implement the case that the requested read actually overlaps with the invalidation in question.

Reply:
Darn. I'm a bit alarmed that this causes the test case that I wrote for `UnstableSymbolic` to pass, and moreover, it seems to do what I'd expect on the SAW side. Perhaps I'm not tickling the right code paths. In any case, I'll look into pushing this logic down lower.

Reply:
I think the logic you have here will work correctly iff you invalidate all the memory in an allocation.  Otherwise, you actually have to check to what degree the requested read overlaps with the invalidation write, which is what the more complicated logic in the body does.

Reply:
Gotcha. Let me know if you'd prefer the test case to distinguish between these two cases of `readMemInvalidate`, but I'm pretty confident that the changes I've just pushed should do the right thing now.